staging/vhost-device-console/README.md

```diff
@@ -0,0 +1,137 @@
+# vhost-device-console - Console emulation backend daemon
+
+## Description
+
+This program is a vhost-user backend that emulates a VirtIO Console device.
+The device's binary takes as parameters a socket path, a socket number which
+is the number of connections, commonly used across all vhost-devices to
+communicate with the vhost-user frontend devices, and the backend type
+"nested" or "network".
+
+The "nested" backend allows input/output to the guest console through the
+current terminal.
+
+The "network" backend creates a local TCP port (specified on vhost-device-console
+arguments) and allows input/output to the guest console via that socket.
+
+This program is tested with QEMU's `vhost-user-device-pci` device.
+Examples' section below.
+
+## Staging Device
+This device will be in `staging` until we complete the following steps:
+- [ ] Support VIRTIO_CONSOLE_F_SIZE feature
+- [ ] Support VIRTIO_CONSOLE_F_EMERG_WRITE feature
```

Comment:
I'm fine to keep the device in staging, but if these features are not mandatory in the spec, we can promote it to the main workspace right now.

I don't have a strong opinion, so it is up to you or someone interested on the device to decide when to move it.

Reply:
Based on the [specs](https://docs.oasis-open.org/virtio/virtio/v1.2/csd01/virtio-v1.2-csd01.html#x1-2960005), I understand that the features: `VIRTIO_CONSOLE_F_SIZE`,
`VIRTIO_CONSOLE_F_MULTIPORT` and `VIRTIO_CONSOLE_F_EMERG_WRITE` are NOT mandatory.
So I also think that we could move this device directly on the main workspace.

Reply:
Maybe one of the reason to keep it in staging is the coverage. Ideally, we would like to avoid decreasing a lot the coverage of the main workspace. So if adding tests it will require some time, maybe we can merge this in staging for now, so you can work on tests in the meantime.

WDYT?

Reply:
Sounds good to me, fully agree on that!

Reply:
@TimosAmpel cool, so can you add something like "Increase test coverage" in this list?
(I hope it's a last change I ask ;-P)

Reply:
No worries at all, I have that added in the list and mark the rest of the bullets as "optional".